docs/designers-developers/developers/internationalization.md

```diff
@@ -1,47 +1,159 @@
+
+## Provide Your Own Translations
+
+You can create and ship your own translations with your plugin, if you have sufficient knowledge of the language(s) you can ensure the translations are available.
+
+### Create Translation File
+
+The translation files must be in the JED 1.x JSON format.
+
+To create a JED translation file, first you need to extract the strings from the text.
+Using the [wp-cli tool](https://wp-cli.org/), you create a `.pot` file using the following command from within your plugin directory:
+
+```
+wp i18n make-pot ./
+```
+
+This will create the file `myguten.pot` a pot file is a template file to be used for translating to different languages. The pot file is made up of translation sets, for example:
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr ""
+```
+
+The `msgid` is the string to be translated, and `msgstr` is the translated string, which in the pot file the msgstr will be empty.
+
+A po file is then created from this template, which is the same format, but the translations filled in. You should copy the file using the language code you are going to translate, this example will use the Esperanto (eo) language:
+
+```
+cp myguten.pot myguten-eo.po
+```
+
+You then go through the `.po` file and add the translation to all the `msgstr` sets:
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr "Simpla bloko"
+```
+
+
+The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you use the [po2json utility](https://github.com/mikeedwards/po2json) which you can install using npm. It might be easiest to install globally using: `npm install -g po2json`. Once installed, use the following command to convert to JED format:
+
+```
+po2json myguten-eo.po myguten-eo.json -f jed
+```
+
+### Load Translation File
+
+The final part is to tell WordPress where it can look to find the translation file. The `wp_set_script_translations` function accepts an optional third argument that is the path it will first check for translations. For example:
+
+```php
+<?php
+	// added in init action
+	wp_set_script_translations( 'myguten-script', 'myguten', plugin_dir_path( __FILE__ ) . 'languages' );
+```
+
+WordPress will check for a file in that path with the format `${domain}-${locale}-${handle}.json` as the source of translations.
+
+So you should create a directory called `languages` and move the `myguten-eo.json` file there, the handle in the file name is the same handle for the script that needs translating, `myguten-script` in this example:
```

Comment:
This is also the place where the POT and PO files should live, so ideally this should be mentioned a bit further up. At least in the examples.

Reply:
I updated and move the directory part up. 👍